bot.py

```diff
@@ -99,6 +110,131 @@ async def get_last_match(ctx, name: str):
     os.remove("df_styled.png")
 
 
+@bot.command(name="add", help="Add the players to the list")
+async def add_summoner(ctx, *, message):
+    """Writes list of summoners to local
+    json file and sends the list to the bot"""
+
+    async with ctx.typing():
+        await asyncio.sleep(1)
+
+    json_path = "data/data.json"
+
+    # create a directory containing json file to store data for added summoners
+    if not os.path.exists(json_path):
+        os.makedirs("data")
+        with open(json_path, "w"):
+            pass
+
+    # accepting comma, space and comma plus space
+    players_list = [x.strip() for x in message.split(",")]
+    players_list = list(filter(None, players_list))
+
+    file_data = ""
+
+    # initializing number of players to display
+    number_of_players = 0
+
+    # initializing server id to a variable
+    server_id = str(ctx.guild.id)
+
+    # storing the json file into a variable
+    if os.path.getsize(json_path) > 0:
+        with open(json_path, "r") as file:
+            file_data = json.load(file)
+
+        # if server id exist in json, add number of players
+        if server_id in file_data:
+            number_of_players += len(file_data[server_id])
+
+    # send error message to bot then exits out of the function if an error with summoner's name
+    for count, _ in enumerate(players_list):
+        if not check_summoner_name(players_list[count]):
+            embed = discord.Embed(
+                title=":x:   Invalid Summoner Name",
+                description=f"`{players_list[count]}` is not a valid summoner name.",
+                color=discord.Color.red(),
+            )
+
+            embed.add_field(
+                name="** **",
+                value="Please enter a valid summoner name!",
+                inline=False,
+            )
+
+            embed.add_field(
+                name="** **",
+                value=f"Adding multiple summoners:\n`@{bot.user.name} add name1, name2`",
+                inline=False,
+            )
+
+            await ctx.send(embed=embed)
+            return None
+
+        # changes the incoming summoner names to what's in the api
+        players_list[count] = get_summoner_name(players_list[count])
```

Comment:
why is this needed?

Reply:
Say if we input a summoner name "lifeissohard", but then the actual league name will be "Life is so hard" and if we want to check if the summoner name exists in our json file, the name has to match.